learn/guides/ballerina-openapi-support.md

```diff
@@ -0,0 +1,314 @@
+8. Add connector icon to the root of the connector. Icon needs to be a `.png` of 200x200 px size with name `icon.png`. 
+9. Update `Ballerina.toml` file referring to the [guide here](https://github.com/ballerina-platform/ballerina-extended-library/discussions/72).
+
+#### **Step 03:** Send PR to openapi-connectors
+1. Run `./gradlew build`. 
+2. Push the changes and create a Pull Request to the master branch of the `openapi-connectors` repository.
+
+## OpenAPI to Ballerina Command Reference
+```bash
+bal openapi [-i | --input] <openapi-contract-file-path> 
+            [-o | --output] <output-location>
+            [--mode <mode-type>]
+            [--tags <tag-names>] 
+            [--operations <operation-names>] 
+            [-n | --nullable]
+            [--license] <license-file-path> 
+            [--with-tests]
+```
+
+- ##### -i | - -input
+Parameter **openapi-contract-path** specifies the path of the OpenAPI contract file (e.g., `my-api.yaml` or `my-api.json`) and is mandatory.
+- ##### -o | - -output
+The Ballerina files will be generated at the same location OpenAPI command is executed. Optionally, you can point to another directory location using the optional flag `(-o|--output)`.
+- ##### - -mode
+Mode type is optional and can be service or client. The Ballerina service and client will be generated according to the mode. Without `--mode` it will generate both service and client stub for the given openAPI contract.
+- ##### - -tags
+To generate the Ballerina client or service stub with a subset of tags defined in the OpenAPI contract, use the options `--tags` and specify the tags you need as specified in the OpenAPI definition.
+```bash
+$ bal openapi -i <openapi contract>  [--tags < "tag1","tag2">]
+```
+- ##### - -operations
+To generate the Ballerina client or service stub with a subset of operations defined in the OpenAPI contract, use the options `--operations` and specify the operations you need as specified in the OpenAPI definition.
+```bash
+$ bal openapi -i <openapi contract> [--operations <"op1", "op2">]
+```
+- ##### - -license
+If you want to generate the Ballerina files with the given copyright or license header, you can use this `--license` flag with your copyright text.
+```bash
+$ bal openapi -i <openapi contract> [--license <license-file-path>]
+```
+- ##### -n |- -nullable
+This is an optional flag in the OpenAPI to Ballerina command. If your OpenAPI specification includes JSON schema properties that are not marked as **nullable:true**, they may return as null in some responses. It will result in a JSON schema to Ballerina record data binding error. If you suspect this can happen for any property, it is safe to generate all data types in the generated record with Ballerina nil support by turning on this flag.
```

Comment:
```suggestion
This is an optional flag in the OpenAPI to Ballerina command. If your OpenAPI specification includes JSON schema properties that are not marked as `nullable:true`, they may be returned as null in some responses. It will result in a JSON schema to Ballerina record data binding error. If you suspect this can happen for any property, it is safe to generate all data types in the generated record with Ballerina nul support by turning on this flag.
```
Is it `nil` or `nul`?

Reply:
afaik both can be used. :)